sql/driver/tx.go

```diff
@@ -17,16 +17,39 @@
 
 package driver
 
-import "fmt"
+import "database/sql/driver"
 
 type tx struct {
 	conn *conn
 }
 
 func (t *tx) Commit() error {
-	return fmt.Errorf("TODO(pmattis): unimplemented")
+	_, err := t.Exec("COMMIT TRANSACTION")
+	return err
 }
 
 func (t *tx) Rollback() error {
-	return fmt.Errorf("TODO(pmattis): unimplemented")
+	_, err := t.Exec("ROLLBACK TRANSACTION")
+	return err
+}
+
+func (t *tx) Exec(query string, args ...interface{}) (driver.Result, error) {
+	var values []driver.Value
```

Comment:
You know the number of values, so it is mildly better to do: `values := make([]driver.Value, 0, len(args))`